Mage/src/main/java/mage/abilities/effects/common/counter/RemoveAllCountersPermanentTargetEffect.java

```diff
@@ -0,0 +1,53 @@
+
+package mage.abilities.effects.common.counter;
+
+import mage.abilities.Ability;
+import mage.abilities.effects.OneShotEffect;
+import mage.constants.Outcome;
+import mage.counters.CounterType;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+
+/**
+ * Removes all counters (optionally of a given counter type) from the target permanent.
+ *
+ * @author MTGfan
+ */
+public class RemoveAllCountersPermanentTargetEffect extends OneShotEffect {
+
+    private final CounterType counterType; // If not null, remove counters of that type only.
+
+    public RemoveAllCountersPermanentTargetEffect() {
+        this((CounterType) null);
+    }
+
+    public RemoveAllCountersPermanentTargetEffect(CounterType counterType) {
+        super(Outcome.Neutral);
+        this.counterType = counterType;
+        staticText = "remove all " + (counterType == null ? "" : counterType.getName() + " ") + "counters from it.";
```

Comment:
basic text having "it" as default target descriptor probably need to change, now that I've added a few cards like Vampire Hexmage to the Effect.

Reply:
yeah, would need an override of getText(mode) and using target description, but with a param (chaining method, or constructor if very common) to use "it" anyway where needed.

Not required for this PR - I don't mind doing it as a followup.